dallinger/registration.py

```diff
@@ -0,0 +1,74 @@
+"""Register experiments through the Open Science Framework."""
+
+import os
+import requests
+
+from dallinger.config import get_config
+
+config = get_config()
+
+root = "https://api.osf.io/v2"
+
+
+def register(dlgr_id, snapshot=None):
+    """Register the experiment."""
+    osf_id = create_osf_project(dlgr_id)
+    upload_assets(dlgr_id, osf_id)
+
+
+def create_osf_project(dlgr_id, description=None):
+    """Create a project on the OSF."""
+
+    osf_access_token = config.get("osf_access_token")
```

Comment:
In anticipation of eventually wanting to allow registration with other services besides OSF, I've adjusted things a bit so that `registration.register` handles the logic of which services to use, calling the OSF-related functions only if this configuration parameter is found.